### PR TITLE
backblaze-b2: 3.2.0 -> 3.5.0

### PR DESCRIPTION
**Pull request by tomhoule**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/backblaze-b2/default.nix [hunk 1]
```diff
@@ -39,6 +41,12 @@ python3Packages.buildPythonApplication rec {
     "test_copy_file_by_id"
     "test_integration"
     "test_get_account_info"
+    "test_passing_api_parameters"
```
Comment (tomhoule) [inline pkgs/development/tools/backblaze-b2/default.nix:44]: This one I don't think requires network, but it appears to expect some file to be there during setup, and it turns out not to be there. I haven't investigated very deeply, but I was unable to figure out if it's a hidden dependency of the test setup, something we could set up, or pointing to an actual bug in the derivation we build.

**Comment by vojta001** [comment]
Result of `nixpkgs-review pr 186684` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>backblaze-b2</li>
  </ul>
</details>

File: pkgs/development/tools/backblaze-b2/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/development/tools/backblaze-b2/default.nix:44]: The issue is that the build environment isn't normally given a "real" `HOME` dir. You can resolve this with:

```
  preCheck = ''
    export HOME=$(mktemp -d)
  '';
```

Comment (tomhoule) [reply pkgs/development/tools/backblaze-b2/default.nix]: That makes sense, thanks for the tip. That indeed helped with this test, and I was even able to remove a few tests from the list of disabled tests. I just pushed the improved commit.

**Review by kevincox — APPROVED** [review]

**Pull request merged**